<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DialogSSHPref</class>
 <widget class="QDialog" name="DialogSSHPref">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>355</width>
    <height>285</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SSH configuration</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QGroupBox" name="gbAuth">
     <property name="title">
      <string>Authentication</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QComboBox" name="comboAuth">
        <item>
         <property name="text">
          <string>Password</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Public key</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Keyboard interactive</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Try all Passwordbased methods</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="stackAuth">
        <property name="frameShape">
         <enum>QFrame::Panel</enum>
        </property>
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="pagePass">
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Username</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="leUsername"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Password</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lePass">
            <property name="echoMode">
             <enum>QLineEdit::Password</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pagePublicKey">
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Select Private Key file</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLineEdit" name="lePrivateKey">
              <property name="placeholderText">
               <string>Path to private key file</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btBrowsePublicKey">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="im.qrc">
                <normaloff>:/im/folder.png</normaloff>:/im/folder.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="flat">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageKeyboard">
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>No parameters needed</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageAllMethods">
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Username and password will be taken from the &quot;Password tab.&quot;</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="gbProtocol">
     <property name="title">
      <string>Protocol type</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QComboBox" name="comboProtocol">
        <item>
         <property name="text">
          <string>Remote shell</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Remote process</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Direct tunnel</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>SFTP</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="stackProtocol">
        <property name="frameShape">
         <enum>QFrame::Panel</enum>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="pageShell">
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>No parameters needed</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageProcess">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Process name</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="leProcName"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageTunnel">
         <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1,0,0">
          <item>
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string extracomment="IP сервера">Local IP</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="leSSHLocalIP">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string extracomment="Адрес сервера">Server IP</string>
            </property>
            <property name="text">
             <string notr="true">127.0.0.1</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>Local port</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="sbSSHLocalPort">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>75</width>
              <height>20</height>
             </size>
            </property>
            <property name="toolTip">
             <string extracomment="Порт сервера">Server port</string>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="maximum">
             <number>99999</number>
            </property>
            <property name="value">
             <number>22</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageSFTP">
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Not realized yet...</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>3</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QPushButton" name="btOK">
       <property name="text">
        <string notr="true">OK</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btCancel">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="im.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>comboAuth</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackAuth</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>81</x>
     <y>34</y>
    </hint>
    <hint type="destinationlabel">
     <x>55</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stackAuth</sender>
   <signal>currentChanged(int)</signal>
   <receiver>comboAuth</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>269</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>269</x>
     <y>44</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboProtocol</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackProtocol</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>77</x>
     <y>175</y>
    </hint>
    <hint type="destinationlabel">
     <x>87</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stackProtocol</sender>
   <signal>currentChanged(int)</signal>
   <receiver>comboProtocol</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>214</x>
     <y>224</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>177</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btCancel</sender>
   <signal>clicked()</signal>
   <receiver>DialogSSHPref</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>241</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>284</x>
     <y>258</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
